go/vt/vtgate/vtgate.go

```diff
@@ -58,60 +58,96 @@ import (
+	// plan cache related flag
+	queryPlanCacheSize   = cache.DefaultConfig.MaxEntries
+	queryPlanCacheMemory = cache.DefaultConfig.MaxMemoryUsage
+	queryPlanCacheLFU    bool
 
-	// Put set-passthrough under a flag.
-	sysVarSetEnabled = flag.Bool("enable_system_settings", true, "This will enable the system settings to be changed per session at the database connection level")
-	setVarEnabled    = flag.Bool("enable_set_var", true, "This will enable the use of MySQL's SET_VAR query hint for certain system variables instead of using reserved connections")
+	maxMemoryRows   = 300000
+	warnMemoryRows  = 30000
+	maxPayloadSize  int
+	warnPayloadSize int
 
-	// lockHeartbeatTime is used to set the next heartbeat time.
-	lockHeartbeatTime = flag.Duration("lock_heartbeat_time", 5*time.Second, "If there is lock function used. This will keep the lock connection active by using this heartbeat")
-	warnShardedOnly   = flag.Bool("warn_sharded_only", false, "If any features that are only available in unsharded mode are used, query execution warnings will be added to the session")
+	noScatter bool
 
-	foreignKeyMode = flag.String("foreign_key_mode", "allow", "This is to provide how to handle foreign key constraint in create/alter table. Valid values are: allow, disallow")
+	// TODO(deepthi): change these two vars to unexported and move to healthcheck.go when LegacyHealthcheck is removed
+
+	// healthCheckRetryDelay is the time to wait before retrying healthcheck
+	healthCheckRetryDelay = 2 * time.Millisecond
+	// healthCheckTimeout is the timeout on the RPC call to tablets
+	healthCheckTimeout = time.Minute
 
-	// flags to enable/disable online and direct DDL statements
-	enableOnlineDDL = flag.Bool("enable_online_ddl", true, "Allow users to submit, review and control Online DDL")
-	enableDirectDDL = flag.Bool("enable_direct_ddl", true, "Allow users to submit direct DDL statements")
+	// System settings related flags
+	sysVarSetEnabled = true
+	setVarEnabled    = true
 
-	enableSchemaChangeSignal = flag.Bool("schema_change_signal", true, "Enable the schema tracker; requires queryserver-config-schema-change-signal to be enabled on the underlying vttablets for this to work")
-	schemaChangeUser         = flag.String("schema_change_signal_user", "", "User to be used to send down query to vttablet to retrieve schema changes")
+	// lockHeartbeatTime is used to set the next heartbeat time.
+	lockHeartbeatTime = 5 * time.Second
+	warnShardedOnly   bool
+
+	// ddl related flags
+	foreignKeyMode     = "allow"
+	dbDDLPlugin        = "fail"
+	defaultDDLStrategy = string(schema.DDLStrategyDirect)
+	enableOnlineDDL    = true
+	enableDirectDDL    = true
+
+	// vtgate schema tracking flags
+	enableSchemaChangeSignal = true
+	schemaChangeUser         string
 )
 
+func RegisterFlags(fs *pflag.FlagSet) {
```

Comment:
does this need to be exported? i can't find anywhere external you call this function